### PR TITLE
Straitjacket Functionality.

### PR DESCRIPTION
**Pull request by brainfood1183**
You can now equip the straitjacket to a humanoid or yourself similarly to equipping a set of handcuffs. The straitjacket is a slower but more effective way of restraining someone.

https://github.com/space-wizards/space-station-14/assets/113240905/a8dfbd5d-85d5-4a2c-8aed-9bad9c8f2382

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Strait Jackets can now be used to restrain individuals.

### CODE REVIEW

**Comment by UbaserB** [comment]
WOOOO LETS GO i needed this functionality

**Comment by UbaserB** [comment]
though, looking at it more closely... I believe that it should take even longer for it to apply considering its not a big big change compared to handcuffs and it takes so long to get out. It also shouldnt hurt since unlike handcuffs its soft fabric; not hard metal. If you really want it to still do damage, it at least wouldn't cause you to bleed or deal bloodloss damage as it'd be closer to brute.

**Review by AJCM-git — CHANGES_REQUESTED** [review]
The only thing this should be changing is making the popup messages configurable (may be too bloaty, idk) or just changing the current "cuffing/uncuffing" messages wording to "restrained/unrestrained" and the like, or something like that, no need for that IsJacket bool

**Comment by UbaserB** [comment]
Another idea to throw out there, is making the strait jacket unescapable but take an extremely long time to put on; so its only reserved for prisoners that want to kill themselves or permas that are disobedient and actively try to escape.

File: Resources/Prototypes/Entities/Objects/Misc/handcuffs.yml [hunk 1]
```diff
@@ -117,3 +117,25 @@
...
+  - type: Handcuff
+    cuffedRSI: Clothing/OuterClothing/Misc/straight_jacket.rsi
+    breakoutTime: 60
+    damageOnResist:
+      types:
+        Blunt: 0
+    cuffTime: 6.5
+    uncuffTime: 5.5
+    stunBonus: 4
+    bodyIconState: body-overlay
```
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Objects/Misc/handcuffs.yml:138]: Isnt this field unnecesary? Also make cuffing/uncuffing take 15 secs and make it size 20

File: Resources/Textures/Clothing/OuterClothing/Misc/straight_jacket.rsi/meta.json [hunk 2]
```diff
@@ -14,6 +14,10 @@
       "name": "equipped-OUTERCLOTHING",
       "directions": 4
     },
+    {
+      "name": "body-overlay-2",
```
Comment (AJCM-git) [inline Resources/Textures/Clothing/OuterClothing/Misc/straight_jacket.rsi/meta.json:18]: is this from tg or another place?

File: Resources/Prototypes/Entities/Objects/Misc/handcuffs.yml [hunk 3]
```diff
@@ -117,3 +117,25 @@
     sprite: Objects/Misc/cablecuffs.rsi
     state: cuff-broken
     color: forestgreen
+
+- type: entity
+  parent: Handcuffs
+  id: ClothingOuterStraightjacket
+  name: straitjacket
+  description: You should probably be wearing this.
```
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Objects/Misc/handcuffs.yml:125]: This should be more descriptive

**Review by AJCM-git — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Objects/Misc/handcuffs.yml (hunk 1 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Misc/handcuffs.yml:138]: The straitjacket functions as an overlay, i think it is needed. Will change all the timings and size to these numbers.

File: Resources/Textures/Clothing/OuterClothing/Misc/straight_jacket.rsi/meta.json (hunk 2 above)
Comment (brainfood1183) [reply Resources/Textures/Clothing/OuterClothing/Misc/straight_jacket.rsi/meta.json:18]: from same place

File: Resources/Prototypes/Entities/Objects/Misc/handcuffs.yml (hunk 3 above)
Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Misc/handcuffs.yml:125]: changed the descripton

**Comment by brainfood1183** [comment]
> Another idea to throw out there, is making the strait jacket unescapable but take an extremely long time to put on; so its only reserved for prisoners that want to kill themselves or permas that are disobedient and actively try to escape.

i think making it ocmpletely unescapable would likely result in a lot of griefing such as  clown putting it on an afk player.

**Comment by UbaserB** [comment]
> > Another idea to throw out there, is making the strait jacket unescapable but take an extremely long time to put on; so its only reserved for prisoners that want to kill themselves or permas that are disobedient and actively try to escape.
> 
> i think making it ocmpletely unescapable would likely result in a lot of griefing such as clown putting it on an afk player.

yeah i suppose thats true, but it still needs a bit of a change

File: Resources/Prototypes/Entities/Objects/Misc/handcuffs.yml (hunk 1 above)
Comment (AJCM-git) [reply Resources/Prototypes/Entities/Objects/Misc/handcuffs.yml:138]: what i meant that the value of bodyIconState already defaults to body-overlay

Comment (AJCM-git) [reply Resources/Prototypes/Entities/Objects/Misc/handcuffs.yml:138]: do test tho

Comment (brainfood1183) [reply Resources/Prototypes/Entities/Objects/Misc/handcuffs.yml]: I removed it, yea works fine.

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**